extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTest.java

```diff
@@ -689,145 +653,11 @@ public void testKillIncompatibleTasks() throws Exception
             null
         )
     ).anyTimes();
-    EasyMock.expect(taskClient.stopAsync("id1", false)).andReturn(Futures.immediateFuture(true));
-    EasyMock.expect(taskClient.stopAsync("id3", false)).andReturn(Futures.immediateFuture(false));
     taskRunner.registerListener(EasyMock.anyObject(TaskRunnerListener.class), EasyMock.anyObject(Executor.class));
-    taskQueue.shutdown("id3", "Task [%s] failed to stop in a timely manner, killing task", "id3");
-
-    EasyMock.expect(taskQueue.add(EasyMock.anyObject(Task.class))).andReturn(true);
-
-    TreeMap<Integer, Map<String, String>> checkpoints = new TreeMap<>();
-    checkpoints.put(0, ImmutableMap.of(
-        shardId0,
-        "0",
-        shardId1,
-        "0"
-    ));
-
-    EasyMock.expect(taskClient.getCheckpointsAsync(EasyMock.contains("id2"), EasyMock.anyBoolean()))
-            .andReturn(Futures.immediateFuture(checkpoints))
-            .times(2);
-
-    replayAll();
-    supervisor.start();
-    supervisor.runInternal();
-    verifyAll();
-  }
-
-  @Test
-  public void testKillBadPartitionAssignment() throws Exception
```

Comment:
Hm, I did some investigation here, and the Kinesis version fails while Kafka passes because the tests had different behavior. 

In the Kafka version of the test, id3-id5 are checking for partition consistency within the partition set (e.g., 0 and 1 can't be in the same set with a taskCount of 2).

In the Kinesis version, there are only two partitions instead of 3, so the case above is tested in id3 only. id4 and id5 (which are duplicates) in kinesis are testing a mismatch of the group ID and the partitions within the group which hits the new check you added.

Reply:
For the kinesis test, I think you could add it back in, remove the redundant "id5" case, and adjust the test to catch the failure on "id4" (might be cool to have the same groupID + partition mismatch test in the kafka version too)

It's definitely out of scope for this PR, but at some point I do want to look into why the kinesis test only uses 2 shards and possibly adjusting that, I think it would be simpler if it followed the structure of the equivalent kafka tests more closely.